lib/engine/train.rb

```diff
@@ -80,15 +90,7 @@ def buyable
     end
 
     def local?
```

Comment:
Remove method. Use local attr.

Reply:
I have to disagree, is it a ruby standard to have a ? in the end when you have a method that will return a boolean?

This was the reason behind the first version of that method.

Reply:
Its true that including a ? in the method is an often used idiom of Ruby; however, the file you are working in isn't using that idiom. See:

      @buyable = true
      @rusted = false
      @obsolete = false
      @operated = false

Prefer conformance to the source, so it has an unified style.

Reply:
if it's just one attr boolean, don't use the ? if it's more complicated use the question mark, in your case i think get rid of init_local

change to

def local?
  return @local if defined?(@local)

  @local = if @distance.is_a?(numeric)
   @distance == 1
else 
  @distance.select { |d| d['nodes'].include?('city') }.sum { | |d| d['visit']  == 1
end

Reply:
Good news is that's what you had and can revert. My read on the style was different than Tobys.